src/main/java/org/opentripplanner/transit/service/TransitService.java

```diff
@@ -94,7 +93,7 @@ Collection<TripPattern> getPatternsForStop(
 
   Map<FeedScopedId, Operator> getOperatorForId();
 
-  Collection<StopLocation> getAllStops();
+  Iterable<StopLocation> getAllStops();
```

Comment:
Even if this is "safe" code, it creates "unsafe" code like this, which is hard to read - hence easy to introduce error in the consumer end. So, I would like a pragmatic approach to this - we return Collection. If we would like to improve on this we can make an interface named "ReadOnlyCollection extends Collection" and use that to document that this is for read only. But, I can at least live with the fact that all read-only services in OTP return `Collection` - and that they are by convention "read-only". Best practice in our service is to use e.g the `List.copyOf()` internally to "freeze" these collections and provoke a runtime error if modified.